python/tchannel/tornado/connection.py

```diff
@@ -294,8 +293,15 @@ def write(self, message):
             message_factory = self.response_message_factory
 
         fragments = message_factory.fragment(message)
+
+        write_future = self._write(fragments.next())
+
         for fragment in fragments:
-            yield self._write(fragment)
+            write_future.add_done_callback(
+                lambda f: f.exception() or self.write(fragment)
```

Comment:
I think this is wrong, I think I need to `set_exception` on `write_future` to propagate the exception correctly.